src/Conversion/ONNXToKrnl/RNN/GRU.cpp

```diff
@@ -602,6 +617,20 @@ void calculateState<GruState, GruActivationPack, GruWeightPack, GruBiasPack>(
           Value nextHt = createMath.add(ztht, ztHt);
 
           // Store the intermediate Ht.
+          // Handle sequence_lens
+          if (!isNoneValue(sequenceLens)) {
+            Value sequenceUB = createKrnl.load(sequenceLens, {bs});
+            Value initial;
+            if (isNoneValue(initialH)) {
+              initial = createMath.constant(nextHt.getType(), 0.);
+            } else {
+              initial = createKrnl.load(initialH, {directionIV, bs, hs});
+            }
+            Value cond = createMath.sge(
+                createMath.cast(sequenceUB.getType(), sequenceIV), sequenceUB);
+            nextHt = createMath.select(cond, /*padding*/ initial, nextHt);
+          }
```

Comment:
Could we create a common function for this to avoid boilerplate? and we can call it in other ops like LSTM and RNN.

Reply:
Changed.